federated-css/e2e/checkNextJsApp.cy.ts

```diff
@@ -0,0 +1,78 @@
+        basePage.openLocalhost(8080);
+    });
+
+    it('Check elements exist', () => {
+        basePage.checkElementContainText({
+            selector: baseSelectors.root,
+            text: Constants.commonPhrases.federatedStyles.header,
+        });
+        basePage.checkElementContainText({
+            selector: baseSelectors.root,
+            text: Constants.commonPhrases.federatedStyles.nextJsAppContainer,
+        });
+    });
+
+    it('Check supported buttons colors', () => {
+        basePage.checkElementVisibility(`${baseSelectors.section}:nth-of-type(1)`);
+
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(1, 1, Constants.color.absoluteBlack));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(1, 2, Constants.color.absolutePurple));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(1, 3, Constants.color.tailwindBlue));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(1, 4, Constants.color.aquamarine));
+    });
+
+    it('Check workaround off', () => {
+        basePage.checkElementVisibility(`${baseSelectors.section}:nth-of-type(2)`);
+
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 1, Constants.color.absoluteBlack));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 2, Constants.color.transparent));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 3, Constants.color.absoluteBlack));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 4, Constants.color.transparent));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 5, Constants.color.transparent));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 6, Constants.color.transparent));
+    });
+
+    it('Check workaround on', () => {
+        cy.get(`${baseSelectors.section}:nth-of-type(2)`)
+          .find(getE2eDataElementByNumber(1))
+          .click();
+        cy.wait(200);
+
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 1, Constants.color.absoluteBlack));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 2, Constants.color.absoluteRed));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 3, Constants.color.absoluteBlack));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 4, Constants.color.absoluteOrange));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 5, Constants.color.brown));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 6, Constants.color.tailwindGreen));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 7, Constants.color.yellow));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 8, Constants.color.tailwindGreen));
+        basePage.checkChildElementHaveProperty(getButtonCompareConfig(2, 9, Constants.color.absoluteBlue));
```

Comment:
please check commonChecks conception. Your checks almost similar, so you can combine it and parse with forEach